File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.js
function solution(arr) {
    let answer = [];
    for(const i of arr){
        if(answer.length != 0 && answer.at(-1) == i){
            answer.pop();
        } else {
            answer.push(i);
        }
    }
    return answer.length == 0 ? [-1] : answer;
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int[] arr) {
        ArrayList<Integer> arrList = new ArrayList<Integer>();
        for(int i = 0; i < arr.length; i++){
            if(!arrList.isEmpty() && arrList.get(arrList.size() - 1) == arr[i]){
                arrList.remove(arrList.size() - 1);
            } else {
                arrList.add(arr[i]);
            }
        }
        if(arrList.isEmpty()) arrList.add(-1);
        int[] answer = arrList.stream().mapToInt(Integer::intValue).toArray();
        return answer;
    }
}